src/routes/v2/analytics.rs

```diff
@@ -0,0 +1,195 @@
+        || allowed_origins.contains(&"*".to_string()))
+    {
+        return Err(ApiError::InvalidInput(
+            "invalid page view URL specified!".to_string(),
+        ));
+    }
+
+    let headers = req
+        .headers()
+        .into_iter()
+        .map(|(key, val)| {
+            (
+                key.to_string().to_lowercase(),
+                val.to_str().unwrap_or_default().to_string(),
+            )
+        })
+        .collect::<HashMap<String, String>>();
+
+    let ip = convert_to_ip_v6(if let Some(header) = headers.get("cf-connecting-ip") {
+        header
+    } else {
+        conn_info.as_deref().unwrap_or_default()
+    })
+    .unwrap_or_else(|_| Ipv4Addr::new(127, 0, 0, 1).to_ipv6_mapped());
+
+    let mut view = PageView {
+        id: Uuid::new_v4(),
+        recorded: Utc::now().timestamp_nanos() / 100_000,
+        domain: domain.to_string(),
+        site_path: url.path().to_string(),
+        user_id: 0,
+        project_id: 0,
+        ip,
+        country: maxmind.query(ip).await.unwrap_or_default(),
+        user_agent: headers.get("user-agent").cloned().unwrap_or_default(),
+        headers: headers
+            .into_iter()
+            .filter(|x| !FILTERED_HEADERS.contains(&&*x.0))
+            .collect(),
+    };
+
+    if let Some(segments) = url.path_segments() {
+        let segments_vec = segments.collect::<Vec<_>>();
+
+        if segments_vec.len() >= 2 {
+            const PROJECT_TYPES: &[&str] = &[
+                "mod",
+                "modpack",
+                "plugin",
+                "resourcepack",
+                "shader",
+                "datapack",
+            ];
+
+            if PROJECT_TYPES.contains(&segments_vec[0]) {
+                let project =
+                    crate::database::models::Project::get(segments_vec[1], &**pool, &redis).await?;
+
+                if let Some(project) = project {
+                    view.project_id = project.inner.id.0 as u64;
+                }
+            }
+        }
+    }
+
+    if let Some((_, user)) = user {
+        view.user_id = user.id.0;
+    }
+
+    analytics_queue.add_view(view).await;
+
+    Ok(HttpResponse::NoContent().body(""))
+}
+
+// TODO: add playtime ingest route
+#[derive(Deserialize)]
+pub struct PlaytimeInput {
+    pub projects: HashMap<crate::models::ids::VersionId, u16>,
+}
+
+#[post("playtime")]
+pub async fn playtime_ingest(
+    req: HttpRequest,
+    maxmind: web::Data<Arc<MaxMindIndexer>>,
+    analytics_queue: web::Data<Arc<AnalyticsQueue>>,
+    session_queue: web::Data<AuthQueue>,
+    playtime_input: web::Json<PlaytimeInput>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<deadpool_redis::Pool>,
+) -> Result<HttpResponse, ApiError> {
+    let (_, user) = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::PERFORM_ANALYTICS]),
+    )
+    .await?;
+
+    // let versions = crate::database::models::Version::get_many(
```

Comment:
I think this shouldnt be commented? Theres a lot of clippy warnings of unused args here